### PR TITLE
trijent shuttle now generic

### PR DESCRIPTION
**Pull request by mullenpaul**
# About the pull request

I have made the Trijent shuttle generic, allowing this to be reused on other mapping projects. We can also have multiple elevators in the same mission.

Trijent dam also now has two elevators.

1. LZ1 to Engineering
2. LZ2 to Omega

# Explain why it's good for the game

Other maps will be able to incorporate the Trijent elevator shuttle.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
refactor: reworked trijent elevators to be reusable on new maps
maptweak: Trijent map now has two elevators, lz1 to engineering and lz2 to omega
/:cl:

### CODE REVIEW

**Comment by mullenpaul** [comment]
Code changes done but I need to check a few things before undrafting

File: code/datums/shuttles.dm [hunk 1]
```diff
@@ -100,6 +100,23 @@
...
+/datum/map_template/shuttle/trijent_elevator/post_load(obj/docking_port/mobile/mobile_dock)
+	if(elevator_network)
+		var/obj/docking_port/mobile/trijent_elevator/elev = mobile_dock
+		elev.elevator_network = elevator_network
+
+/datum/map_template/shuttle/trijent_elevator/A
+	elevator_network="A"
+
+/datum/map_template/shuttle/trijent_elevator/B
+	elevator_network="B"
```
Comment (harryob) [inline code/datums/shuttles.dm:122]:
```suggestion
	elevator_network = "A"

/datum/map_template/shuttle/trijent_elevator/B
	elevator_network = "B"
```

File: code/modules/shuttle/computers/trijent_elevator_control.dm [hunk 2]
```diff
@@ -38,8 +43,15 @@
 
 /obj/structure/machinery/computer/shuttle/elevator_controller/proc/get_landing_zones()
 	. = list()
+	var/obj/docking_port/mobile/trijent_elevator/shuttle = SSshuttle.getShuttle(shuttleId)
+
 	for(var/obj/docking_port/stationary/trijent_elevator/elev in SSshuttle.stationary)
-		. += list(elev)
+		if(shuttle.elevator_network == "" || elev.elevator_network == null)
```
Comment (harryob) [inline code/modules/shuttle/computers/trijent_elevator_control.dm:49]:
```suggestion
		if(!shuttle.elevator_network)
```

File: code/modules/shuttle/shuttles/trijent_elevator.dm [hunk 3]
```diff
@@ -74,6 +76,17 @@
 	door_control.control_doors("force-lock-launch")
 	qdel(door_control)
 
+/obj/docking_port/stationary/trijent_elevator/occupied
+	name="occupied"
+	id=STAT_TRIJENT_OCCUPIED
+	airlock_exit="west"
```
Comment (harryob) [inline code/modules/shuttle/shuttles/trijent_elevator.dm:82]:
```suggestion
	name = "occupied"
	id = STAT_TRIJENT_OCCUPIED
	airlock_exit = "west"
```

File: code/modules/shuttle/shuttles/trijent_elevator.dm [hunk 4]
```diff
@@ -74,6 +76,17 @@
...
+	roundstart_template = /datum/map_template/shuttle/trijent_elevator
+
+/obj/docking_port/stationary/trijent_elevator/empty
+	name="empty"
+	id=STAT_TRIJENT_EMPTY
+	airlock_exit="west"
```
Comment (harryob) [inline code/modules/shuttle/shuttles/trijent_elevator.dm:88]:
```suggestion
	name = "empty"
	id = STAT_TRIJENT_EMPTY
	airlock_exit = "west"
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by fira — APPROVED** [review]

**Comment by mullenpaul** [comment]
This is working on TM

**Review by harryob — APPROVED** [review]

**Pull request merged**